src/structured_data.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use rustc_serialize::{Encodable, Encoder, Decoder};
+use cbor;
+use error::RoutingError;
+use NameType;
+use sodiumoxide::crypto;
+use std::str;
+
+/// StructuredData
+/// These types may be stored unsigned with previous and current owner keys
+/// set to the same keys. Updates though require a signature to validate
+#[derive(Clone, RustcDecodable, RustcEncodable)]
+pub struct StructuredData {
```

Comment:
<2> Ah no, NameType is just temporarily there (as i am currently not using the sodiumoxide fork). It'll be Digest eventually. Even then i guess we shouldn't assume utf8 for it can come from any source, not necessarily hash of utf8 encoded string.

<4> Ok

<5> Ok too. This was just about naming. Most of the time we will not be transferring ownership and simply be having `previous_owner_keys` field empty and dealing with signatures of `current_owner_keys` only to update the StructuredData. So it seemed strange to call it `previous_owner_signatures`. But that's just naming, so it's Ok.